File: Leetcode/70. Climbing Stairs/solution.rb
def climb_stairs(n)
  steps = (0..n).to_a

  steps[0] = 1
  steps[1] = 1

  (2..n).each { |i| steps[i] = steps[i - 1] + steps[i - 2] }

  steps[n]
end

puts climb_stairs(7)

File: Leetcode/70. Climbing Stairs/solution.js
const climbStairs = (n) => {
  const steps = new Array(n+1)

  steps[0] = 1
  steps[1] = 1

  for(let i = 2; i <= n; i++){
    steps[i] = steps[i - 1] + steps[i - 2]
  }

  return steps[n]
}

console.log(climbStairs(7))